src/legacy/core_plugins/vis_type_vega/public/components/vega_vis_editor.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiCodeEditor } from '@elastic/eui';
+import compactStringify from 'json-stringify-pretty-compact';
+// @ts-ignore
+import hjson from 'hjson';
+import { i18n } from '@kbn/i18n';
+
+import { toastNotifications } from 'ui/notify';
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import { VisParams } from '../vega_fn';
+import { VegaHelpMenu } from './vega_help_menu';
+import { VegaActionsMenu } from './vega_actions_menu';
+
+const aceOptions = {
+  maxLines: Infinity,
+  highlightActiveLine: false,
+  showPrintMargin: false,
+  tabSize: 2,
+  useSoftTabs: true,
+  wrap: true,
+};
+
+function format(value: string, stringify: typeof compactStringify) {
+  try {
+    const spec = hjson.parse(value, { legacyRoot: false, keepWsc: true });
+    return stringify(spec);
```

Comment:
I don't think we can skip options like it was before https://github.com/elastic/kibana/pull/47473/files#diff-2abd10d2a2ed5649151d832d6ae32983L63. 
For example without options reformatting as HJON removes comments:
![Oct-14-2019 16-00-46](https://user-images.githubusercontent.com/11524637/66753161-d3aa3680-ee9b-11e9-9991-e69c19e2e592.gif)